### PR TITLE
Update to ginkgo v2 and remove deprecated ginkgo reporter

### PR DESCRIPTION
**Pull request by KevinJCross**
our project the cf [app-autoscaler-release](https://github.com/cloudfoundry/app-autoscaler-release) needs to move to ginkgo v2 
Unfortunately the way it is used means that v1 and v2 can not exist in the same dependency tree.
This means that the test module in this project needs to be dealt with ... the easiest thing is to remove
This would be a breaking change.
Its likely if there is no reponse we will move to a newer logging library.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+golang 1.17.7
```
Comment (ctlong) [inline .tool-versions:1]: Can you please remove this file.

File: lager_suite_test.go [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
+	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"runtime"
 
 	"testing"
 )
 
+const MaxThreads = 100
+var _ = BeforeSuite(func() {
+	runtime.GOMAXPROCS(MaxThreads)
```
Comment (ctlong) [inline lager_suite_test.go:13]: Why is this necessary?

**Review by ctlong — CHANGES_REQUESTED** [review]
@KevinJCross I've left some comments. In order to have this merged I think you'd also have to rebase these commits off of develop and change the base branch for the PR.

File: .tool-versions (hunk 1 above)
Comment (KevinJCross) [reply .tool-versions:1]: Have you ever used asdf ... this is a file for asdf to define the sdk's you are using. 
I would recomend keeping this as its useful for anyone using this library and does not hurt anyone.

File: lager_suite_test.go (hunk 2 above)
Comment (KevinJCross) [reply lager_suite_test.go:13]: If I remember correctly it was to enable more threads to reduce the runtime of the tests ... its been a while now.

Comment (KevinJCross) [reply lager_suite_test.go:13]: It also has been moved from [writer_sink_test.go](https://github.com/cloudfoundry/lager/pull/54/files/a053a8ef0561e8279b52a1dfe161ce61a92d6d3d#diff-f43d3d0989e10a86b7c8fa0eb6eb9ab4627cbbfcc8fc70a326794771033e5c18)

File: .tool-versions (hunk 1 above)
Comment (KevinJCross) [reply .tool-versions:1]: tbh ... its very useful for golang too.
The workflows can check the compatibility (we do this in autoscaler where we use the mod file to determine the go version and update the asdf file) but the asdf sets the golang that is currently been used for developement. 
Its obviously a personal choice for ease of use so Ill remove it as requested.

**Review by ctlong — APPROVED** [review]
